### PR TITLE
What's Here for BasicObject

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Not the most important What's Here section, but I want to build many such so that there's a tree of linked What's Here sections.  Object, for example, will link to (as "what's elsewhere") the What's Here sections of both Kernel and BasicObject.

### CODE REVIEW

File: object.c [hunk 1]
```diff
@@ -4488,6 +4488,30 @@ f_sprintf(int c, const VALUE *v, VALUE _)
...
  *      end
  *    end
+ *
+ *  === What's Here
+ *
+ *  These are the methods defined for \BasicObject:
+ *
+ *  - ::new:: Returns a new \BasicObject instance.
+ *  - {!}[#method-i-21]:: Returns the boolean negation of +self+: +true+ or +false+.
+ *  - {!=}[#method-i-21-3D]:: Returns +self+ and the given object are _not_ equal.
```
Comment (jeremyevans) [inline object.c:4498]: `Returns` -> `Returns whether`

File: object.c [hunk 2]
```diff
@@ -4488,6 +4488,30 @@ f_sprintf(int c, const VALUE *v, VALUE _)
...
+ *  - {==}[#method-i-3D-3D]:: Returns whether +self+ and the given object
+ *                            are the same object.
+ *  - {__id__}[#method-i-__id__]:: Returns the integer object identifier for +self+.
+ *  - {__send__}[#method-i-__send__]:: Invokes the method identied by the given symbol.
```
Comment (jeremyevans) [inline object.c:4502]: `Invokes` -> `Calls`.  Invoke isn't wrong, but calling a method is the term more frequently used.

File: object.c [hunk 3]
```diff
@@ -4488,6 +4488,30 @@ f_sprintf(int c, const VALUE *v, VALUE _)
...
+ *
+ *  === What's Here
+ *
+ *  These are the methods defined for \BasicObject:
+ *
+ *  - ::new:: Returns a new \BasicObject instance.
+ *  - {!}[#method-i-21]:: Returns the boolean negation of +self+: +true+ or +false+.
+ *  - {!=}[#method-i-21-3D]:: Returns +self+ and the given object are _not_ equal.
+ *  - {==}[#method-i-3D-3D]:: Returns whether +self+ and the given object
+ *                            are the same object.
```
Comment (jeremyevans) [inline object.c:4500]: `are the same object` -> `are equivalent`.  `==` is very different from `equal?` and should not have the same description.

File: object.c [hunk 4]
```diff
@@ -4488,6 +4488,30 @@ f_sprintf(int c, const VALUE *v, VALUE _)
...
+ *  - {__id__}[#method-i-__id__]:: Returns the integer object identifier for +self+.
+ *  - {__send__}[#method-i-__send__]:: Invokes the method identied by the given symbol.
+ *  - #equal?:: Returns whether +self+ and the given object are the same object.
+ *  - #instance_eval:: Evaluates the given string or block.
```
Comment (jeremyevans) [inline object.c:4504]: ... `in the context of +self+`.  The important difference of `instance_eval` and regular `eval` is that it changes the context.

File: object.c [hunk 5]
```diff
@@ -4488,6 +4488,30 @@ f_sprintf(int c, const VALUE *v, VALUE _)
...
+ *  - #instance_exec:: Executes the given block, passing it the given arguments.
```
Comment (jeremyevans) [inline object.c:4505]: `Executes the given block in the context of +self+, passing the given arguments`

File: object.c [hunk 6]
```diff
@@ -4488,6 +4488,30 @@ f_sprintf(int c, const VALUE *v, VALUE _)
...
+ *  - #method_missing:: Specifies the method to be called when +self+ receives
+ *                      a message it cannot handle.
```
Comment (jeremyevans) [inline object.c:4507]: `Method called when an undefined method is called on +self+`

File: object.c [hunk 7]
```diff
@@ -4488,6 +4488,30 @@ f_sprintf(int c, const VALUE *v, VALUE _)
...
+ *  - #singleton_method_added:: Specifies the method to be called when a singleton
```
Comment (jeremyevans) [inline object.c:4508]: `Method called when a singleton method is added to +self+.` Please use similar wording for `singleton_method_removed` and `singleton_method_undefined`

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thank you for working on this.  Please see the inline comments.

**Comment by BurdetteLamar** [comment]
Note: no TOC needed for this one.

**Review by jeremyevans — APPROVED** [review]
Looks good!

**Pull request merged**